src/mlpack/methods/ann/layer/linear3d_impl.hpp

```diff
@@ -0,0 +1,178 @@
+/**
+ * @file methods/ann/layer/linear3d_impl.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * Implementation of the Linear layer class which accepts 3D input.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_LINEAR3D_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_LINEAR3D_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "linear3d.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType,
+    typename RegularizerType>
+Linear3D<InputDataType, OutputDataType, RegularizerType>::Linear3D() :
+    inSize(0),
+    outSize(0)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType,
+    typename RegularizerType>
+Linear3D<InputDataType, OutputDataType, RegularizerType>::Linear3D(
+    const size_t inSize,
+    const size_t outSize,
+    RegularizerType regularizer) :
+    inSize(inSize),
+    outSize(outSize),
+    regularizer(regularizer)
+{
+  weights.set_size(outSize * inSize + outSize, 1);
+}
+
+template<typename InputDataType, typename OutputDataType,
+    typename RegularizerType>
+void Linear3D<InputDataType, OutputDataType, RegularizerType>::Reset()
+{
+  typedef typename arma::Mat<typename OutputDataType::elem_type> MatType;
+
+  weight = MatType(weights.memptr(), outSize, inSize, false, false);
+  bias = MatType(weights.memptr() + weight.n_elem, outSize, 1, false, false);
+}
+
+template<typename InputDataType, typename OutputDataType,
+    typename RegularizerType>
+template<typename eT>
+void Linear3D<InputDataType, OutputDataType, RegularizerType>::Forward(
+    const arma::Mat<eT>& input, arma::Mat<eT>& output)
+{
+  typedef typename arma::Mat<eT> MatType;
+  typedef typename arma::Cube<eT> CubeType;
+
+  if (input.n_rows % inSize != 0)
+  {
+    Log::Fatal << "Number of features in the input must be divisible by inSize."
+               << std::endl;
+  }
+
+  const size_t nPoints = input.n_rows / inSize;
+  const size_t batchSize = input.n_cols;
+
+  output.set_size(nPoints * outSize, batchSize);
+
+  CubeType inputTemp(const_cast<MatType&>(input).memptr(), nPoints, inSize,
+      batchSize, false, false);
+
+  for (size_t i = 0; i < batchSize; ++i)
+  {
+    //! Shape of weight : (outSize, inSize).
+    //! Shape of inputTemp : (nPoints, inSize, batchSize).
```

Comment:
Doxygen ignores comments that are unrelated to declarations.
```suggestion
    // Shape of weight : (outSize, inSize).
    // Shape of inputTemp : (nPoints, inSize, batchSize).
```